### DISCUSSION TITLE
Upgrading from 11.0.2 to 11.0.10 results in transparent window and high memory usage

### DISCUSSION THREAD

**Discussion by alexhooperist** [AvaloniaUI/Avalonia#16027] [Q&A]
### Describe the bug

I upgraded just avalonia via NuGet from v11.0.2 to 11.0.10 and suddenly my window appears transparent
![image](https://github.com/AvaloniaUI/Avalonia/assets/141257217/5555c9fc-e054-40f0-bafa-c28b627e3c70)

And my memory usage skyrocketed. It was sitting at 148mb max usage before upgrading, and is now at 7.4gb. This screenshot is the memory of the app just sitting still.
![image](https://github.com/AvaloniaUI/Avalonia/assets/141257217/44ae9075-e13e-4a14-8536-7f8c47c880d7)

### To Reproduce

1. Create a C# Avalonia .NET MVVM App in Visual studio.
2. Run the app.
3. Downgrade to 11.0.2.
4. Run the app.
5. Upgrade to 11.0.10.
6. Run again.

### Avalonia version

11.0.10

### OS

Windows

### Additional context

Avalonia.Diagnostics, Avalonia.Fonts.Inter, Avalonia.ReactiveUI, and Avalonia.Themes.Fluent were also available but I did not install them. Same bug occurs when I do install them.

Project is based on the visual studio C# Avalonia.Net MVVM 

My visual studio version is 17.10.0 Preview

**Comment by thevortexcloud** [comment]
I could be wrong (I don't use the Visual Studio package manager often), but doesn't your screenshot show you still have packages that can be upgraded?

**Comment by alexhooperist** [comment]
> I could be wrong (I don't use the Visual Studio package manager often), but doesn't your screenshot show you still have packages that can be upgraded?

Same bug occurs when they are installed. They were Avalonia.Diagnostics, Avalonia.Fonts.Inter, Avalonia.ReactiveUI, and Avalonia.Themes.Fluent. I left them uninstalled so I could isolate the issue to just the Avalonia nuget package.

**Comment by timunie** [comment]
we need a minimal sample to check.

**Reply by alexhooperist** [reply]
https://github.com/alexhooperist/Avalonia-Bug-Sample

**Reply by maxkatz6** [reply]
So, why are you mixing Avalonia versions in the same project?

**Reply by maxkatz6** [reply]
There is no guarantee that different avalonia versions would work together:
```
[Visual]Exception when rendering: 'System.TypeLoadException: Method 'get_PublicFeatures' in type 'Avalonia.Skia.SkiaContext' from assembly 'Avalonia.Skia, Version=11.0.2.0, Culture=neutral, PublicKeyToken=c8d484a7012f9a8b' does not have an implementation.
   at Avalonia.Skia.PlatformRenderInterface.CreateBackendContext(IPlatformGraphicsContext graphicsContext)
   at Avalonia.Rendering.PlatformRenderInterfaceContextManager.EnsureValidBackendContext()
   at Avalonia.Rendering.Composition.Server.ServerCompositor.RenderCore()'(ServerCompositor #54089615)
```

There is no bug here.

**Reply by maxkatz6** [reply]
We might need to implement some additional guard to show compile time error, when different versions are used together.

**Reply by alexhooperist** [reply]
Like I said, the bug doesn't go anywhere when I update everything else too.

**Reply by alexhooperist** [reply]
I pushed a change to the above repo with all the other packages updated to 11.0.10. Same bug. 

Although after letting it sit it does finally say Welcome to avalonia.

Memory usage still through the roof and a transparent window for about 30 seconds.

**Reply by maxkatz6** [reply]
No issues here when everything is updated. Try to delete bin/obj folders and rebuild everything, though I didn't need to do that.

![image](https://github.com/AvaloniaUI/Avalonia/assets/3163374/65164acb-1018-480f-8177-85bf5aa6de73)

**Reply by alexhooperist** [reply]
OK, this is weird.
I upgraded everything to 11.0.11 and that magically fixed it all.
I then downgraded back to 11.0.10 and it was still fixed.

**Reply by alexhooperist** [reply]
> No issues here when everything is updated. Try to delete bin/obj folders and rebuild everything, though I didn't need to do that.
> 
> ![image](https://private-user-images.githubusercontent.com/3163374/340810599-65164acb-1018-480f-8177-85bf5aa6de73.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTg3NDA0OTgsIm5iZiI6MTcxODc0MDE5OCwicGF0aCI6Ii8zMTYzMzc0LzM0MDgxMDU5OS02NTE2NGFjYi0xMDE4LTQ4MGYtODE3Ny04NWJmNWFhNmRlNzMucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDYxOCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA2MThUMTk0OTU4WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MzEzNDk3NWRiNTAwODk1NzMyYjkxZDRjNTM1ZGM0MWRjMmMwOTVmODVhNGFjYTI3NTBhNWUyMTRkMzljY2I4NCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.KAEkPFGtyoyjFOUvqPjbeaHpvDlvPRR_bhUnNdyAV4I)

I will try this too.

**Reply by alexhooperist** [reply]
> > No issues here when everything is updated. Try to delete bin/obj folders and rebuild everything, though I didn't need to do that.
> > ![image](https://private-user-images.githubusercontent.com/3163374/340810599-65164acb-1018-480f-8177-85bf5aa6de73.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTg3NDA0OTgsIm5iZiI6MTcxODc0MDE5OCwicGF0aCI6Ii8zMTYzMzc0LzM0MDgxMDU5OS02NTE2NGFjYi0xMDE4LTQ4MGYtODE3Ny04NWJmNWFhNmRlNzMucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDYxOCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA2MThUMTk0OTU4WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MzEzNDk3NWRiNTAwODk1NzMyYjkxZDRjNTM1ZGM0MWRjMmMwOTVmODVhNGFjYTI3NTBhNWUyMTRkMzljY2I4NCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.KAEkPFGtyoyjFOUvqPjbeaHpvDlvPRR_bhUnNdyAV4I)
> 
> I will try this too.

This doesn't appear to have done anything. The 11.0.11 fix did it though so whatever.

**Reply by alexhooperist** [reply]
Thanks for the help!

**Comment by liu4113231988** [comment]
The same problem.

**Reply by maxkatz6** [reply]
Check logs
